Vagrant fedora machine does not come up

# Description of problem

With the draft pull request I was able to bring up a Fedora based VM for running CI tests using `vagrant up fedora`.  However with the merged version of the vagrant patches, attempting to bring up the Fedora VM no longer works (the Ubuntu based one seems to get further).

# Expected result

```
$ vagrant up fedora
$ vagrant ssh fedora
```

Can then use the fedora VM.

# Actual result

```
$ vagrant up fedora
[ . . . ]
$ vagrant ssh fedora
==> fedora: Running action triggers before ssh ...
==> fedora: Running trigger...
==> fedora: Domain is not created. Please run `vagrant up` first.
```

# Further information

Full transcript to follow.

> So,
> 
> > Is there a variable or some other mean of not force Podman with Fedora?
> 
> you want to use neither or use Docker instead (like it was before)? I'm curious what's happening with Podman that breaks it.

I revert the patch and it started to work; that made me think it was related with Podman. I wanted to double-check that, so I need a way of setting `USE_DOCKER=yes` and this really be honored even if in Fedora.

Btw, that's the error that I am getting:
```
[install_kata_image.sh:42] INFO: Building image with AGENT_INIT=no
Creating rootfs for ubuntu
/home/vagrant/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/rootfs.sh -o 2.2.0-rc0-f3539d2fb7668aa557fead555a5fff6854f723eb-dirty -r /home/vagrant/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/ubuntu_rootfs ubuntu
INFO: rootfs_lib.sh file found. Loading content
INFO: Detecting agent rust version
INFO: Get rust version from /home/vagrant/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/../../../versions.yaml
Required rust version: 1.47.0
INFO: Detecting musl version
INFO: Get musl version from /home/vagrant/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/../../../versions.yaml
Required musl version: 1.1.23
/home/vagrant/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/ubuntu /home/vagrant/go/src/github.com/kata-containers/kata-containers/tools/osbuilder
/home/vagrant/go/src/github.com/kata-containers/kata-containers/tools/osbuilder
STEP 1: FROM docker.io/ubuntu:20.04
Getting image source signatures
Copying blob 16ec32c2132b done  
Copying config 1318b700e4 done  
Writing manifest to image destination
Storing signatures
STEP 2: RUN apt-get update && apt-get --no-install-recommends install -y     apt-utils     autoconf     automake     binutils     build-essential     ca-certificates     chrony     coreutils     curl     debianutils     debootstrap     g++     gcc     git     libc6-dev     libstdc++-8-dev     m4     make     musl     musl-dev     musl-tools     protobuf-compiler     sed     systemd     tar     vim     wget
2021-08-30T21:13:01.000430303Z: sd-bus call: Transport endpoint is not connected
error running container: error creating container for [/bin/sh -c apt-get update && apt-get --no-install-recommends install -y     apt-utils     autoconf     automake     binutils     build-essential     ca-certificates     chrony     coreutils     curl     debianutils     debootstrap     g++     gcc     git     libc6-dev     libstdc++-8-dev     m4     make     musl     musl-dev     musl-tools     protobuf-compiler     sed     systemd     tar     vim     wget]: : exit status 1
Error: error building at STEP "RUN apt-get update && apt-get --no-install-recommends install -y     apt-utils     autoconf     automake     binutils     build-essential     ca-certificates     chrony     coreutils     curl     debianutils     debootstrap     g++     gcc     git     libc6-dev     libstdc++-8-dev     m4     make     musl     musl-dev     musl-tools     protobuf-compiler     sed     systemd     tar     vim     wget": error while running runtime: exit status 1
make: *** [Makefile:93: /home/vagrant/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/.ubuntu_rootfs.done] Error 125
[install_kata_image.sh:50] ERROR: sudo -E USE_DOCKER= DISTRO=ubuntu make -e image
[install_kata.sh:22] ERROR: .ci/install_kata_image.sh 

```

--- a/.ci/setup_env_ubuntu.sh
+++ b/.ci/setup_env_ubuntu.sh
@@ -102,6 +102,12 @@ main()
 		pkgs_to_install+=" ${rust_agent_pkgs[@]}"
 	fi
 
+	# The redis-server package fails to install if IPv6 is disabled. Let's
+	# check if that's the case and then enable it.
+	if [ $(sudo sysctl -n net.ipv6.conf.all.disable_ipv6) -eq 1 ]; then
+		sudo sysctl -w net.ipv6.conf.all.disable_ipv6=0
+	fi
+
 	chronic sudo -E apt -y install $pkgs_to_install
 
 	[ "$setup_type" = "minimal" ] && exit 0

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -12,6 +12,9 @@
 job = ENV['CI_JOB'] || ""
 guest_user = 'vagrant'
 guest_home_dir = '/home/vagrant'
+# The file on the guest where environment variables are going to be set
+# to export.
+guest_env_file = guest_home_dir + '/ci_job_env'
 
 # All Vagrant configuration is done below. The "2" in Vagrant.configure
 # configures the configuration version (we support older styles for
@@ -48,7 +51,7 @@ Vagrant.configure("2") do |config|
     chown -R #{guest_user}:#{guest_user} "${GOPATH}"
     kata_tests_repo_dir="${GOPATH}/src/github.com/kata-containers/tests"
 
-    env_file="#{guest_home_dir}/ci_job_env"
+    env_file="#{guest_env_file}"
     sudo -E PATH=$PATH -H -u #{guest_user} \
     cat <<-EOF > ${env_file}
 export GOPATH="$GOPATH"
@@ -71,12 +74,6 @@ EOF
 
     distro=$(source /etc/os-release; echo $ID)
     case "$distro" in
-      "fedora")
-        # Fedora >= 32 Kernel comes configured with cgroup v2 by default.
-        # This switches back to cgroup v1. It requires a reboot.
-        sudo dnf install -y grubby
-        sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
-        ;;
       "ubuntu")
         # TODO: redis-server package fails to install if IPv6 is disabled. Move this
         # code to the setup script.
@@ -86,12 +83,6 @@ EOF
         fi
         ;;
     esac
-
-    # Build the osbuilder with same distro as the host.
-    export osbuilder_distro="$distro"
-
-    cd ${kata_tests_repo_dir}
-    sudo -E PATH=$PATH -H -u #{guest_user} bash -c '.ci/setup.sh'
   SHELL
 
   config.vm.define "fedora", autostart: false do |fedora|
@@ -99,11 +90,28 @@ EOF
     # Fedora is required to reboot so that the change to cgroups v1
     # makes effect.
     fedora.vm.provision "shell", reboot: true, inline: <<-SHELL
-      echo "Need to reboot the VM"
+      sudo dnf install -y grubby
+      # Set the kernel parameter to use cgroups v1.
+      sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
+    SHELL
+
+    fedora.vm.provision "shell", inline: <<-SHELL
+      source "#{guest_env_file}"
+      cd "${GOPATH}/src/github.com/kata-containers/tests"
+      # Build the osbuilder with same distro as the host.
+      export osbuilder_distro="fedora"
+      sudo -E PATH=$PATH -H -u #{guest_user} bash -c '.ci/setup.sh'
     SHELL
   end
 
   config.vm.define "ubuntu", autostart: false do |ubuntu|
     ubuntu.vm.box = "generic/ubuntu2004"
+    ubuntu.vm.provision "shell", inline: <<-SHELL
+      source "#{guest_env_file}"
+      cd "${GOPATH}/src/github.com/kata-containers/tests"
+      # Build the osbuilder with same distro as the host.
+      export osbuilder_distro="ubuntu"
+      sudo -E PATH=$PATH -H -u #{guest_user} bash -c '.ci/setup.sh'
+    SHELL
   end
 end

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -71,18 +71,6 @@ EOF
     echo "cd $kata_tests_repo_dir" >> #{guest_home_dir}/.bashrc
     sudo -E PATH=$PATH -u #{guest_user} \
     echo "source $env_file" >> #{guest_home_dir}/.bashrc
-
-    distro=$(source /etc/os-release; echo $ID)
-    case "$distro" in
-      "ubuntu")
-        # TODO: redis-server package fails to install if IPv6 is disabled. Move this
-        # code to the setup script.
-        if [[ $(sysctl -n net.ipv6.conf.all.disable_ipv6) -eq 1 ]]; then
-          sudo sysctl -w net.ipv6.conf.all.disable_ipv6=0
-          sudo sed -i 's/\(net.ipv6.conf.all.disable_ipv6\).*/\1=0/' /etc/sysctl.conf
-        fi
-        ;;
-    esac
   SHELL
 
   config.vm.define "fedora", autostart: false do |fedora|

